### PR TITLE
sd: Return ret if error in sdmmc_spi_read_cxd

### PR DESCRIPTION
**Pull request by alexsven**
- Add return ret if CMD fails in spi_read

### CODE REVIEW

**Review by carlescufi — DISMISSED** [review]

**Review by danieldegrasse — DISMISSED** [review]
Does this fix the issue described in #55769 on your board?

**Comment by alexsven** [comment]
> Does this fix the issue described in #55769 on your board?

Kind of, it returns back to the inital init function with the -ENOTSUP as it should, but the CS pin is never released, causing problems other places, so there is more going on here.

File: subsys/sd/sd_ops.c [hunk 1]
```diff
@@ -214,6 +214,7 @@ static int sdmmc_spi_read_cxd(struct sd_card *card, uint32_t opcode, uint32_t *c
 	ret = sdhc_request(card->sdhc, &cmd, &data);
 	if (ret) {
 		LOG_DBG("CMD%d failed: %d", opcode, ret);
+		return ret;
```
Comment (jfischer-no) [inline subsys/sd/sd_ops.c:217]: If it is changed to be handled as an error, then the logging message above should also be changed to LOG_ERR().

Comment (alexsven) [reply subsys/sd/sd_ops.c:217]: It's not in the function below either, but I do agree.

**Comment by alexsven** [comment]
I added a check of the OCR that is read by using CMD58. If OCR == 0 we should reject the card as no operating conditions are set. I have verified that this fixes https://github.com/zephyrproject-rtos/zephyr/issues/55769 and have also verified that with an SD card inserted it passes and works as normal

File: subsys/sd/sd_ops.c (hunk 1 above)
Comment (alexsven) [reply subsys/sd/sd_ops.c:217]: I can remove this again if needed, as we really shouldn't get to this point with an invalid SD card.

File: subsys/sd/sdmmc.c [hunk 2]
```diff
@@ -79,6 +79,11 @@ static int sdmmc_spi_send_ocr(struct sd_card *card, uint32_t arg)
 	ret = sdhc_request(card->sdhc, &cmd, NULL);
```
Comment (koffes) [inline subsys/sd/sdmmc.c:79]: I see that from before, this ret value was not checked.

**Review by koffes — CHANGES_REQUESTED** [review]

File: subsys/sd/sd_ops.c (hunk 1 above)
Comment (alexsven) [reply subsys/sd/sd_ops.c]: Removed, to not muddle the PR

**Review by koffes — APPROVED** [review]

**Comment by alexsven** [comment]
@danieldegrasse or @jfischer-no: Do you see any issues with checking the OCR value and rejecting the card if no values are set?

**Comment by danieldegrasse** [comment]
> @danieldegrasse or @jfischer-no: Do you see any issues with checking the OCR value and rejecting the card if no values are set?

It should be ok to do this- the OCR value should always have some voltage range value set. I've tested this PR on a K64F and it functions correctly.

**Review by danieldegrasse — APPROVED** [review]

**Review by jfischer-no — APPROVED** [review]

**Pull request merged**